bitbots_vision_tools/scripts/extract_from_rosbag.py

```diff
@@ -0,0 +1,115 @@
+#!/usr/bin/env python
+
+import rosbag
+import argparse
+import os
+from sensor_msgs.msg import Image
+import cv_bridge
+import cv2
+from cv_bridge import CvBridge
+import numpy as np
+
+try:
+    input = raw_input
+except NameError:
+    pass
+
+parser = argparse.ArgumentParser("Extract images from a rosbag")
+
+parser.add_argument("-o", "--out-dir", required=True, help="Output directory for the images", dest="outputdir")
+parser.add_argument("-i", "--input-bag", required=True, help="Input rosbag", dest="inputfile")
+parser.add_argument("-n", "--nth-image", help="Every n-th image will be saved, prompted if not specified", dest="n",
+                    type=int)
+parser.add_argument("-t", "--topic", help="Image topic, prompted if not asked", dest="topic")
+
+args = parser.parse_args()
+try:
+    bag = rosbag.Bag(args.inputfile, "r")
+except IOError:
+    print("error opening bag")
+    exit(1)
+
+topics_and_type = bag.get_type_and_topic_info()
+image_topics_and_info = []
+
+for topic, info in topics_and_type.topics.items():
+    if info.msg_type == "sensor_msgs/Image":
+        image_topics_and_info.append([topic, info])
+
+print(image_topics_and_info)
+if len(image_topics_and_info) == 0:
+    print("No messages of type sensor_msgs/Image found in the provided rosbag")
+    exit()
+elif len(image_topics_and_info) == 1:
+    print(
+        "Found exactly one topic ({0}) of type sensor_msgs/Image with {1} messages".format(image_topics_and_info[0][0],
+                                                                                           image_topics_and_info[0][
+                                                                                               1].message_count))
+    if image_topics_and_info[0][0] == args.topic:
+        chosen_set_num = 0
+    else:
+        selection = input("Do you want to extract images from this topic? y/n: ")
+        if selection != "y":
+            exit()
+        chosen_set_num = 0
+else:
+    print("Multiple topics with type sensor_msgs/Image:")
+    specified_topic_in_topics = False
+    for i, topic_touple in enumerate(image_topics_and_info):
+        print("[" + str(i) + "] topic: " + str(topic_touple[0]) + " \t message_count: " + str(
+            topic_touple[1].message_count))
+        if topic_touple[0]:
+            chosen_set_num = i
+            specified_topic_in_topics = True
+    if not specified_topic_in_topics:
+        try:
+            chosen_set_num = int(input("Make a selection [0-{}] or q: ".format(len(image_topics_and_info) - 1)))
+        except ValueError:
+            exit()
+
+chosen_set = image_topics_and_info[chosen_set_num]
+
+print("The dataset you have selected has a frequency of {0}".format(chosen_set[1].frequency))
+if args.n is None:
+    try:
+        n = int(input("Every n-th image will be saved. Please specify n:"))
+    except ValueError:
+        exit()
+else:
+    n = args.n
+
+print("Extracting every {}th image".format(n))
```

Comment:
Kind of ugly for 1, 2 and 3

Reply:
@timonegk, please specify your comment

Reply:
"Extracting every 1th / 2th / 3th image" is not a correct sentence, but we can keep it that way